libsolidity/interface/CompilerStack.cpp

```diff
@@ -1502,7 +1502,12 @@ void CompilerStack::compileContract(
 
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
 
-	std::shared_ptr<Compiler> compiler = std::make_shared<Compiler>(m_evmVersion, m_revertStrings, m_optimiserSettings);
+	std::shared_ptr<Compiler> compiler = std::make_shared<Compiler>(
+		m_evmVersion,
+		m_eofVersion,
+		m_revertStrings,
+		m_optimiserSettings
+	);
```

Comment:
`Compiler` and `CompilerContext` are the two places where it's ok to hard-code `nullopt` for EOF version. They do use `ABIFunctions` and `YulUtilFunctions` but only in legacy context.

On the other hand it's not wrong, just unnecessary so I think we can leave it at that and just merge the PR.